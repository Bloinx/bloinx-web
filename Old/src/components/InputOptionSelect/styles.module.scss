.InputOptionSelect {
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
}

.InputOptionSelectLabel {
  font-size: 12px;
  color: rgba($color: #000000, $alpha: 0.54);
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  margin-bottom: 8px;
}
