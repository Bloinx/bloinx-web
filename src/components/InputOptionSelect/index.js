import React from "react";
import classnames from "classnames";
import PropTypes from "prop-types";
import Typography from "@mui/material/Typography";

import styles from "./styles.module.scss";

export default function InputOptionSelect({
  label,
  value,
  options,
  onChange,
  name,
}) {
  const handleOnClick = (value) => {
    onChange({
      target: {
        value,
        name,
      },
    });
  };

  return (
    <div className={styles.InputOptionSelect}>
      <Typography
        variant="body1"
        component="div"
        className={styles.InputOptionSelectLabel}
      >
        {label}
      </Typography>

      <div className={styles.InputOptionSelectOptions}>
        {options.map((option) => {
          return (
            <button
              key={option.label}
              type="button"
              onClick={() => handleOnClick(option.value)}
              className={classnames(
                styles.InputOptionSelectOptionsButton,
                option.value === value && styles.InputOptionSelectOptionsActive
              )}
            >
              {option.label}
            </button>
          );
        })}
      </div>
    </div>
  );
}

InputOptionSelect.defaultProps = {
  label: "",
  value: "",
};

InputOptionSelect.propTypes = {
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.bool,
  ]),
  options: PropTypes.instanceOf(Array).isRequired,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
};
