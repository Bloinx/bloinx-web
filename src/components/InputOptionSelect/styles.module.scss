@import "../../theme.scss";

.InputOptionSelect {
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
  margin-top: 2%;
  margin-bottom: 2%;
}

.InputOptionSelectLabel {
  margin-top: 2%;
  margin-bottom: 2%;
}

.InputOptionSelectOptions {
  display: flex;
  margin-top: 2%;
  justify-content: center;
}

.InputOptionSelectOptionsButton {
  margin-right: 10px;
  height: 40px;
  min-width: 35px;
  background-color: $gray;
  border: none;
  border-radius: 3px;
  color: white;
  font-size: 16px;
  cursor: pointer;
}

.InputOptionSelectOptionsActive {
  background-color: $pink;
}
