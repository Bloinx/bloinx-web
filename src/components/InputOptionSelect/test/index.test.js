import React from "react";
import renderer from "react-test-renderer";

import InputOptionSelect from "../index";

describe("InputOptionSelect", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <InputOptionSelect
          name="some-name"
          value="1"
          label="some-label"
          options={[]}
          onChange={jest.fn()}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
