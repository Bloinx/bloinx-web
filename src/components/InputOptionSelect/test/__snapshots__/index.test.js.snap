// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputOptionSelect renders correctly 1`] = `
<span
  className="InputOptionSelect"
>
  <span
    className="ant-typography InputOptionSelectLabel"
    style={
      Object {
        "WebkitLineClamp": undefined,
      }
    }
  >
    some-label
  </span>
  <div
    className="ant-radio-group ant-radio-group-solid"
  />
</span>
`;
